lms/views/api/sync.py

```diff
@@ -48,8 +49,19 @@ def sync(request):
             grading_student_id=grading_student_id,
         )
 
+    # Sync the groups over to H so they are ready to be annotated against
     request.find_service(name="lti_h").sync(
         groupings, request.parsed_params["group_info"]
     )
+
+    # Store the relationship between the assignment and the groupings
+    assignment_service = request.find_service(name="assignment")
+    assignment_service.upsert_assignment_groupings(
+        assignment=assignment_service.get_assignment_by_id(
```

Comment:
`upsert_assignment_groupings` uses the non ORM bulk upsert so it only uses `assignmen.id` 

We could simplify the PR based on that and save one precious DB query. I'm also happy with the approach of using the whole object shown here.

Reply:
Yeah. I think this is worth doing. We've chatted about this and decided this can be merged to get the stats, and then we can follow up with a fix.